tests/Helper/RulesNormalizerTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Tests\Helper;
+
+use PHPUnit\Framework\TestCase;
+use ReflectionProperty;
+use Yiisoft\Validator\Helper\RulesNormalizer;
+use Yiisoft\Validator\Tests\Support\Data\ObjectWithDifferentPropertyVisibility;
+
+final class RulesNormalizerTest extends TestCase
+{
+    public function dataNormalize(): array
+    {
+        return [
+            'null' => [[], null],
```

Comment:
(Optional) I'd add a comment (with link possibly) that more cases are covered in `Validator`.

Reply:
https://github.com/yiisoft/validator/pull/394/files#r1033426303

Reply:
done